lib/auto_attach.py

```diff
@@ -98,8 +98,9 @@ def main(cfg: UAConfig):
             )
         )
         return 1
+    finally:
+        system.remove_file(AUTO_ATTACH_STATUS_MOTD_FILE)
```

Comment:
But then we would also need to remove the return from the generic `Exception` right ? That's why I added the `finally` statement here

Reply:
The `Exception` case already has line 89: `system.remove_file(AUTO_ATTACH_STATUS_MOTD_FILE)` so it doesn't need to fall through and can return early.

But we could also change it to not return early and just set a `ret = 1` variable to be returned later.

I think the thing I'd like to avoid is combining the use of early returns and `finally` because it's behavior isn't always intuitive/obvious.

Reply:
Ohh my mistake, I missed that generic `Exception` was removing that MOTD file as well. Let's just remove the return as you mentioned

Reply:
@lucasmoura sorry for the confusion here (all due to my original mistake leaving this return here :upside_down_face:) - we want the disable_auto_attach case to return zero

Reply:
Right, because this is the systemd script we don't want systemd to report degraded on a failure for this reason IMO. I think the original cli auto-attach returned 0 for this reason too?

Reply:
Yes, indeed. Okay, I will just remove the return then. Thanks for the whole context @orndorffgrant